server/cluster/cluster.go

```diff
@@ -1690,30 +1693,59 @@ func (c *RaftCluster) SetMinResolvedTS(storeID, minResolvedTS uint64) error {
 	return c.putStoreLocked(newStore)
 }
 
-func (c *RaftCluster) runMinResolvedTSJob(saveInterval time.Duration) {
+func (c *RaftCluster) isLarger(minResolvedTSRealtime uint64) bool {
+	c.RLock()
+	defer c.RUnlock()
+	return minResolvedTSRealtime != math.MaxUint64 && minResolvedTSRealtime > c.minResolvedTS
+}
+
+func (c *RaftCluster) runMinResolvedTSJob() {
+	defer logutil.LogPanic()
 	defer c.wg.Done()
-	if saveInterval == 0 {
-		return
+
+	interval := c.opt.GetMinResolvedTSPersistenceInterval()
+	if interval == 0 {
+		interval = DefaultMinResolvedTSPersistenceInterval
 	}
-	defer logutil.LogPanic()
-	ticker := time.NewTicker(saveInterval)
+	ticker := time.NewTicker(interval)
 	defer ticker.Stop()
+
+	c.LoadMinResolvedTS()
 	for {
 		select {
 		case <-c.ctx.Done():
 			log.Info("min resolved ts background jobs has been stopped")
 			return
 		case <-ticker.C:
-			minResolvedTS := c.GetMinResolvedTS()
-			if minResolvedTS != math.MaxUint64 {
-				c.Lock()
-				c.storage.SaveMinResolvedTS(minResolvedTS)
-				c.Unlock()
+			interval = c.opt.GetMinResolvedTSPersistenceInterval()
+			if interval != 0 {
+				minResolvedTSRealtime := c.GetMinResolvedTS()
+				if c.isLarger(minResolvedTSRealtime) {
+					c.Lock()
+					c.minResolvedTS = minResolvedTSRealtime
+					c.storage.SaveMinResolvedTS(minResolvedTSRealtime)
+					c.Unlock()
+				}
+			} else {
+				interval = DefaultMinResolvedTSPersistenceInterval
```

Comment:
So when the interval is set to zero we will skip the check?

Reply:
yes, we will not persist the value.